staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -3075,6 +3075,62 @@ type ContainerStatus struct {
 	// +featureGate=SupplementalGroupsPolicy
 	// +optional
 	User *ContainerUser `json:"user,omitempty" protobuf:"bytes,13,opt,name=user,casttype=ContainerUser"`
+	// AllocatedResourcesStatus represents the status of various resources
+	// allocated for this Pod.
+	// +featureGate=ResourceHealthStatus
+	// +optional
+	// +patchMergeKey=name
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=name
+	AllocatedResourcesStatus []ResourceStatus `json:"allocatedResourcesStatus,omitempty" patchStrategy:"merge" patchMergeKey:"name" protobuf:"bytes,14,rep,name=allocatedResourcesStatus"`
+}
+
+type ResourceStatus struct {
+	Name ResourceName `json:"name" protobuf:"bytes,1,opt,name=name"`
```

Comment:
If every ResouceHealth has a unique ID, what is this name?  What data format is it?  How long can it be?  Is it required or optional? Does it need to be unique in some context?

Since it is a listmap key it must be `+required` - please say that.

Reply:
added both - required and validation